pkg/manifestclient/read_roundtripper.go

```diff
@@ -6,21 +6,30 @@ import (
 	"fmt"
 	"io"
 	"io/fs"
-	apidiscoveryv2 "k8s.io/api/apidiscovery/v2"
-	"k8s.io/apimachinery/pkg/util/json"
 	"net/http"
 	"strconv"
 	"strings"
 	"sync"
 	"time"
 
+	apidiscoveryv2 "k8s.io/api/apidiscovery/v2"
+	"k8s.io/apimachinery/pkg/util/json"
+
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
 	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/features"
 	"k8s.io/apiserver/pkg/server"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
+func init() {
+	// This feature gate is needed to set requestInfo.LabelSelector
+	utilruntime.Must(utilfeature.DefaultMutableFeatureGate.Set(fmt.Sprintf("%s=true", features.AuthorizeWithSelectors)))
```

Comment:
maybe we should add a new unit test in a new package that would simply use the manifest client with a label selector ? 

for example: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/features/testing/features_init_test.go